packages/beacon-node/test/unit/chain/archiveStore/utils/xdelta3.test.ts

```diff
@@ -0,0 +1,143 @@
+      state.balances.set(0, state.balances.get(0) + 1000);
+      state.commit();
+      return ssz.phase0.BeaconState.serialize(state.toValue() as BeaconState<ForkName.phase0>);
+    },
+  },
+  {
+    title: "Sepolia state",
+    base: () => {
+      return Buffer.from(
+        fs.readFileSync(path.join(import.meta.dirname, "../../../../fixtures/binaryDiff/source.txt"), "utf8"),
+        "hex"
+      );
+    },
+    changed: () => {
+      return Buffer.from(
+        fs.readFileSync(path.join(import.meta.dirname, "../../../../fixtures/binaryDiff/input.txt"), "utf8"),
+        "hex"
+      );
+    },
+  },
+];
+
+const binaryValue = (s: string): Uint8Array => Uint8Array.from(Buffer.from(s, "utf8"));
+
+describe("BinaryDiffCodec", () => {
+  let codec: IStateDiffCodec;
+  let multiDiffData: Record<string, {value: Uint8Array; diff: Uint8Array}>;
+
+  beforeAll(async () => {
+    codec = new XDelta3Codec();
+
+    multiDiffData = {
+      snapshot: {
+        value: binaryValue("initial value"),
+        diff: Buffer.alloc(0),
+      },
+      diff1: {
+        value: binaryValue("initial value + diff 1"),
+        diff: codec.compute(binaryValue("initial value"), binaryValue("initial value + diff 1")),
+      },
+      diff2: {
+        value: binaryValue("initial value + diff 1 + diff 2"),
+        diff: codec.compute(binaryValue("initial value + diff 1"), binaryValue("initial value + diff 1 + diff 2")),
+      },
+      diff3: {
+        value: binaryValue("initial value + diff 1 + diff 2 + diff 3"),
+        diff: codec.compute(
+          binaryValue("initial value + diff 1 + diff 2"),
+          binaryValue("initial value + diff 1 + diff 2 + diff 3")
+        ),
+      },
+    };
+  });
+
+  it.each(testsCases)("$title", ({base, changed}) => {
+    const _base = base();
+    const _changed = changed();
+
+    const delta = codec.compute(_base, _changed);
+    const result = codec.apply(_base, delta);
+
+    expect(delta).toBeInstanceOf(Uint8Array);
+    expect(delta).not.toHaveLength(0);
+    expect(Buffer.from(result).toString("hex")).toStrictEqual(Buffer.from(_changed).toString("hex"));
+  });
+
+  describe("multiple diffs", () => {
+    it("should produce valid value for one diff", () => {
+      expect(codec.apply(multiDiffData.snapshot.value, multiDiffData.diff1.diff)).toEqual(multiDiffData.diff1.value);
+    });
+
+    it("should produce valid value for two diffs", () => {
+      expect(
+        codec.apply(codec.apply(multiDiffData.snapshot.value, multiDiffData.diff1.diff), multiDiffData.diff2.diff)
+      ).toEqual(multiDiffData.diff2.value);
+    });
+
+    it("should produce valid value for three diffs", () => {
+      expect(
+        codec.apply(
+          codec.apply(codec.apply(multiDiffData.snapshot.value, multiDiffData.diff1.diff), multiDiffData.diff2.diff),
+          multiDiffData.diff3.diff
+        )
+      ).toEqual(multiDiffData.diff3.value);
+    });
+  });
+});
+
+function attestationFromValues(
```

Comment:
Why not reuse ‘phase0SingleAttestationFromValues’ from sszBytes.test.ts?

Reply:
We enforce a linting rule to `no exports in tests files`, to have no dependencies among test (*.test.ts)  files. So can not reuse that function here and was not in favor to refactor all such functions into some utils file.